api/app/wildfire_one.py

```diff
@@ -393,26 +385,54 @@ async def fetch_hourlies(
 
 
 async def get_hourly_readings(
+        session: ClientSession,
+        header: dict,
         station_codes: List[int],
-        time_of_interest: datetime) -> List[WeatherStationHourlyReadings]:
+        start_timestamp: datetime,
+        end_timestamp: datetime) -> List[WeatherStationHourlyReadings]:
     """ Get the hourly readings for the list of station codes provided.
     """
     # Create a list containing all the tasks to run in parallel.
     tasks = []
 
-    # Limit the number of concurrent connections.
-    conn = TCPConnector(limit=10)
-    async with ClientSession(connector=conn) as session:
-        # Get the authentication header
-        header = await _get_auth_header(session)
+    # Iterate through "raw" station data.
+    iterator = _fetch_raw_stations_generator(
+        session, header, BuildQueryByStationCode(station_codes))
+    async for raw_station in iterator:
+        task = asyncio.create_task(
+            fetch_hourlies(session,
+                           raw_station,
+                           header,
+                           start_timestamp,
+                           end_timestamp))
+        tasks.append(task)
 
-        # Iterate through "raw" station data.
-        iterator = _fetch_raw_stations_generator(
-            session, header, BuildQueryByStationCode(station_codes))
-        async for raw_station in iterator:
+    # Run the tasks concurrently, waiting for them all to complete.
+    return await asyncio.gather(*tasks)
+
+
+async def get_hourly_readings_all_stations(
```

Comment:
Uh oh - this is going to create hundreds of simultaneous connections to the WF1 api - I think we have to rethink this. That's probably why you're hitting those limits - we need to reduce the number of calls here.

Reply:
Actually this prevented issues, otherwise we run into sessions being closed.